### ISSUE TITLE
Destructure object

### ISSUE THREAD

**Issue by fabien0102** [nicoespeon/abracadabra#208]
## Is this request related to a problem? Please describe.

My classic flow to create a react component is the following one:
- Create a file with the correct name
- Use the following vscode snippet
```json
{  
  "Create a new react component": {
    "scope": "typescript",
    "prefix": "newReactComponent",
    "body": [
      "import React from \"react\";",
      "",
      "export interface ${TM_FILENAME_BASE}Props {$1}",
      "",
      "export const ${TM_FILENAME_BASE} = (props: ${TM_FILENAME_BASE}Props) => {$2};"
    ],
    "description": "Create a new react component"
  }
}
```
In real life, `MyComponent.tsx` file will result to this file:

```tsx
import React from "react";

interface MyComponentProps {
  // Cursor here
}

export const MyComponent = (props: MyComponentProps) => ();
```

- Now it's time to think about the component props, taking time to create a pleasant API
- And here, is my "problem"!

I have this file:
```tsx
import React from "react";

interface MyComponentProps {
  name: string;
  age: number;
}

export const MyComponent = (props: MyComponentProps) => ();
```

And usually want to destructure my `props` inside my component to avoid to write `props.{myProps}` everytime. You probably guess it, I need to do this manually, and if I have a lot of props, this is boring!

## Describe the solution you'd like

On the following snippet:
```tsx
import React from "react";

interface MyComponentProps {
  name: string;
  age: number;
}

export const MyComponent = (props: MyComponentProps) => ();
```

I want to put my cursor on `props`, ask for `Destructure object` and boom!

```diff
- export const MyComponent = (props: MyComponentProps) => ();
+ export const MyComponent = ({ name, age }: MyComponentProps) => ();
```

## Additional context

To complete this challenge, with style, if I did already used `props.name` and I don't have any name conflicts, I would love to spread the refactor 😍

So:
```diff
- export const MyComponent = (props: MyComponentProps) => (<div>{props.name} is {props.age} years old</div>);
+ export const MyComponent = ({ name, age }: MyComponentProps) => (<div>{name} is {age} years old</div>);
```

**Comment by nicoespeon** [comment]
I found an extension that seems to do that: https://twitter.com/_paulshen/status/1392914417243672578?s=20

Note that I didn't try it. Don't know how many edge-cases it handles.

**Comment by nicoespeon** [comment]
Implemented together with #414 

Will be shipped in the next release (in the following days) 🚀
